static/js/public/snap-details/map.js

```diff
@@ -0,0 +1,121 @@
+/* global d3, topojson */
+
+export default function renderMap(el, snapData, isPublic) {
+  const mapEl = d3.select(el);
+
+  d3.queue()
+    .defer(d3.json, "/static/js/world-110m.v1.json")
+    .await(ready);
+
+  function getColourScaleModifier(value) {
+    let colorId = ~~(value * 10);
+    if (colorId > 9) { colorId = 9; } // so that 100% doesn't go out of scale
+
+    return '--scale-' + colorId;
+  }
+
+  function render(mapEl, snapData, world) {
+    const width = mapEl.property('clientWidth');
+    const height = width * 0.5;
+    // some offset position center of the map properly
+    const offset = width * 0.1;
+
+    const projection = d3.geoNaturalEarth1()
+      .scale(width * 0.2)
+      .translate([(width / 2), ((height + offset) / 2) ])
+      .precision(.1);
+
+    // rotate not to split Asia
+    projection.rotate([-10, 0]);
+
+    const path = d3.geoPath()
+      .projection(projection);
+
+    // clean up HTML before rendering map
+    mapEl.html('');
+
+    const svg = mapEl.append("svg")
+      .attr("width", width)
+      .attr("height", height);
+
+    const tooltip = mapEl.append("div")
+      .attr("class", "snapcraft-territories__tooltip u-no-margin");
+
+    const tooltipMsg = tooltip.append("div")
+      .attr("class", "p-tooltip__message");
+
+    const countries = topojson.feature(world, world.objects.countries).features;
+
+    const g = svg.append("g");
+    const country = g.selectAll(".snapcraft-territories__country").data(countries);
+
+    country.enter().insert("path")
+      .attr("class", countryData => {
+        const countrySnapData = snapData[countryData.id];
+
+        if (countrySnapData) {
+          return `snapcraft-territories__country snapcraft-territories__country${getColourScaleModifier(
+            countrySnapData.percentage_of_users
+          )}`;
+        }
+
+        return 'snapcraft-territories__country';
+      })
+      .attr("d", path)
+      .attr("id", function(d) {
+        return d.id;
+      })
+      .attr("title", function(d) {
+        return d.properties.name;
+      })
+      .on("mousemove", countryData => {
+        const pos = d3.mouse(mapEl.node());
+        const countrySnapData = snapData[countryData.id];
+        let className = ['snapcraft-territories__swatch'];
+
+        if (countrySnapData) {
+          className.push(`snapcraft-territories__swatch${getColourScaleModifier(
+            countrySnapData.percentage_of_users
+          )}`);
+
+          tooltip
+            .style('top', pos[1] + 'px')
+            .style('left', pos[0] + 'px')
+            .style('display', 'block');
+
+          let content = ['<span class="u-no-margin--top">', countrySnapData.name];
+          if (!isPublic) {
```

Comment:
As per comment above, we could possibly not need this flag and check if `number_of_users` exists in data and if not show simplified (public) tooltip, otherwise show the data for publisher in the tooltip.
If the data for public users contains `number_of_users` it's a bug, as we shouldn't expose it in any form.